dockstore-webservice/src/main/java/io/dockstore/webservice/core/CollectionEntry.java

```diff
@@ -26,6 +26,8 @@ public class CollectionEntry implements Serializable {
     @JsonProperty("categories")
     private List<CategorySummary> categorySummaries = new ArrayList<>();
 
+    public CollectionEntry() {}
```

Comment:
What's the purpose of this new constructor?

Reply:
Think @hyunnaye messaged about this, she was running into 
```
2024-04-23T21:06:32.040Z
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.dockstore.webservice.core.CollectionEntry` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 43] (through reference chain: io.dockstore.webservice.core.Collection["entries"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67)
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1915)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:414)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1360)
```

That said, this constructor does seem like it would leave the collectionentry in an un-usable state and should throw an exception and/or be a starting point for using a debugger to figure out a better approach.

Reply:
This is to fix the endpoint `curl -X POST -H 'Authorization: Bearer <redacted>' -H 'Content-Type: application/json' -d '{"name": "A-0", "topic": "", "entries": [{}]}' https://qa.dockstore.org/api/organizations/0/collections`.

It returns a 500 due to a `InvalidDefinitionException` because `CollectionEntry` doesn't have a default constructor.